### ISSUE TITLE
Kernel connection with Windows 10 WSL

### ISSUE THREAD

**Issue by aminya** [nteract/hydrogen#1914]
Currently, Hydrogen isn't able to connect to the kernels that are running locally by the WSL:

Using the following remote server config
```
[{
  "name": "WSL server",
  "options": { "baseUrl": "http://127.0.0.1:8888/" }
}]
```
It asks for the token, and after authentication it fails to detect kernel specs.

**Comment by aviatesk** [comment]
Did you follow the instructions [here](https://nteract.gitbooks.io/hydrogen/docs/Usage/RemoteKernelConnection.html) ? You need to set authentication token in the config as well.
I'm able to connect to e.g. IHaskell kernel in WSL.

**Comment by aminya** [comment]
Yes. I tried both hard coding the token, and also giving it when Atom asks for.

**Comment by rlevy-fn** [comment]
A small gotcha that's tripped me up with this in the past -- the grammar selected by Atom for the active file needs to match an installed kernel spec. If you try to connect while viewing a new file, ctrl-shift-L & select python _before_ connecting Hydrogen.
